modules/app_aks/main.tf

```diff
@@ -32,6 +32,15 @@ resource "azurerm_kubernetes_cluster" "default" {
     load_balancer_sku = "standard"
   }
 
+  dynamic "microsoft_defender" {
+    for_each = var.use_azure_defender ? [1] : []
+    content {
+      log_analytics_workspace_id = var.log_analytics_workspace_id
+    }
+  }
+
+  workload_identity_enabled = false
```

Comment:
I'm only defaulting explicitly to avoid the drift. The default is `false` in the provider, but if we don't set it's `null`, so next run Azure will already set it as a `false` which will cause drift.